src/config/secret.config.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  GetSecretValueCommand,
+  SecretsManagerClient
+} from '@aws-sdk/client-secrets-manager';
+
+import mailConfig from '../config/mail.config';
+
+const client = new SecretsManagerClient({ region: mailConfig.awsRegion });
+
+const command = new GetSecretValueCommand({
+  SecretId: mailConfig.secretManagerName
+});
+
+const getSecretValue = async () => {
+  try {
+    const data = await client.send(command);
+
+    return JSON.parse(data.SecretString);
+  } catch (error) {
+    console.error(error);
+  }
```

Comment:
No need of `try...catch`. Let's the error be thrown it there is any error.

Reply:
if we throw error here then i don't think code will reach  ` 'x-api-key':
        secretValue['MAIL_SERVER_PASSWORD'] || mailConfig.mailServer.password` here.

Reply:
Handle it where the function is called, not here where function is declared. There is not way to find out if error occurred from where the function is called if we capture the error on function declaration